bip-0360.mediawiki

```diff
@@ -0,0 +1,596 @@
+algorithm. This new address type protects transactions submitted to the mempool and helps preserve the free market by
+preventing the need for private, out-of-band mempool transactions.
+
+The following table is non-exhaustive but intended to inform the average Bitcoin user whether their bitcoin is
+vulnerable to a long-range quantum attack:
+
+{| class="wikitable"
+|+ Vulnerable address types
+|-
+! Address type !! Vulnerable !! Prefix !! Example
+|-
+| P2PK || Yes || 04 ||
+0496b538e853519c726a2c91e61ec11600ae1390813a627c66fb8be7947be63c52da7589379515d4e0a604f8141781e62294721166bf621e73a82cbf
+2342c858ee
+|-
+| P2PKH || No || 1 || 1A1zP1eP5QGefi2DMPTfTL5SLmv7DivfNa
+|-
+| P2WPKH || No || bc1q || bc1qsnh5ktku9ztqeqfr89yrqjd05eh58nah884mku
+|-
+| P2TR || Yes || bc1p || bc1p92aslsnseq786wxfk3ekra90ds9ku47qttupfjsqmmj4z82xdq4q3rr58u
+|}
+
+It should be noted that Taproot addresses are vulnerable in that they encode a 32-byte x-only public key, from which a
+full public key can be reconstructed.
+
+If a key is recovered by a CRQC it can also be trivially checked to see if any child keys were produced using an
+unhardened [https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki BIP-32] derivation path.
+
+==== Long Range and Short Range Quantum Attacks ====
+
+Long Range Quantum Attack is an attack in which the public key has been exposed on the blockchain for an extended
+period of time, giving an attacker ample opportunity to break the cryptography. This affects:
+
+* Early addresses (Satoshi's coins, CPU miners, starts with 04)
+* Reused addresses (any type, except P2QRH)
+* Taproot addresses (starts with bc1p)
+* Unhardened BIP-32 HD wallet keys
+
+Short Range Quantum Attack is an attack that must be executed quickly while a transaction is still in the mempool,
+before it gets mined into a block. This affects:
+
+* Any transaction in the mempool (except for P2QRH)
+* Unhardened BIP-32 HD wallet keys
+
+Short-range attacks require much larger, more expensive CRQCs since they must be executed within the short window
+before a transaction is mined. Long-range attacks can be executed over a longer timeframe since the public key remains
+exposed on the blockchain indefinitely.
+
+Should quantum advantage manifest, a convention is proposed in spending the full 65-byte P2PK key used by the coinbase
+output in Block 1 back to itself. It is proposed to call the address in Block 1 the
+[https://mempool.space/address/0496b538e853519c726a2c91e61ec11600ae1390813a627c66fb8be7947be63c52da7589379515d4e0a604f81
+41781e62294721166bf621e73a82cbf2342c858ee
+Canary address] since it can only be spent from by others (assuming Satoshi's continued absence) if secp256k1 is
+broken. Should the Canary coins move, that will signal that reliance on secp256k1 is presently vulnerable. Without the
+Canary, or an address like it, there may be some doubt as to whether the coins were moved with keys belonging to the
+original owner.
```

Comment:
This still leaves in the possibility that Satoshi comes back and spends the coins. Why don't we choose coins that we know nobody knows the private key for, i.e. coins locked to a NUMS point?

Reply:
Is there an example of that? If so, a link would be very helpful.

Reply:
I only managed to find two non-empty addresses that use an obvious NUMS point containing around 0.02 BTC in total:

- [1D8eDztgv79J59V7UBBpNGnRE6hjstqKb5](https://mempool.space/address/1D8eDztgv79J59V7UBBpNGnRE6hjstqKb5), whose uncompressed public key's x-coordinate is the SHA-256 hash of the secp256k1 generator point in uncompressed form (see [BIP341](https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki#constructing-and-spending-taproot-outputs)).
- [1JRy3CrZciF42nqTJESVuqsBXU3z2p1mDJ](1JRy3CrZciF42nqTJESVuqsBXU3z2p1mDJ), whose uncompressed public key's x-coordinate is 1.

Given this lack of funds locked in NUMS addresses and Murch's other criticisms (which I agree with) I'd recommend completely removing the concept of canary coins from the BIP.